.github/workflows/snyk_scan.yaml

```diff
@@ -0,0 +1,30 @@
+---
+name: vanagon check PRs
+
+on: [pull_request]
+
+jobs:
+ snyk_vanagon:
+   runs-on: ubuntu-latest
+   steps:
+    - name: checkout the current PR
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+    - name: Run Vanagon Snyk Scan
+      uses: puppetlabs/security-snyk-vanagon-action@v2.2.0
+      with:
+        snykToken: ${{ secrets.SNYK_PE_TOKEN }}
+        snykOrg: 'puppet-enterprise'
+        skipPlatforms: 'debian-10-armhf'
+        urlsToReplace: 'artifactory.delivery.puppetlabs.net,%s/xart,builds.delivery.puppetlabs.net,%s/xbuild'
+        newHost: 'localhost'
+        rproxyUser: ${{ secrets.RPROXY_USER }}
+        rproxyKey: ${{ secrets.RPROXY_KEY }}
+        sshKey: ${{ secrets.SEC_SECBOT_SSH_PUBLIC }}
```

Comment:
`SEC_PUBLIC_RPROXY_USER` is now available
`SEC_PUBLIC_RPROXY_KEY` is now available
`SEC_SECBOT_SSH_PUBLIC` is now available

Reply:
Thanks @LivingInSyn Unfortunately we're seeing a warning:

https://github.com/puppetlabs/pxp-agent-vanagon/runs/5101556125?check_suite_focus=true#step:4:12

```
/usr/local/bin/start_script.sh: line 21: warning: command substitution: ignored null byte in input
```

And then a fatal error trying to retrieve data from remapped build server URL:

https://github.com/puppetlabs/pxp-agent-vanagon/runs/5101556125?check_suite_focus=true#step:4:40

```
2022/02/07 23:26:29 Got bad output from vanagon dependencies on pxp-agent redhatfips-7-x86_64. Try #0 Output: Error loading project 'pxp-agent' using '/github/workspace/configs/projects/pxp-agent.rb':
Error: 401. Unable to get source from http://localhost/xbuild/puppet-runtime/202202020/artifacts/agent-runtime-main-202202020.redhatfips-7-x86_64.settings.yaml.sha1
```

It looks like the project is trying to inherit settings from the `pxp-agent` component in https://github.com/puppetlabs/pxp-agent-vanagon/blob/328192431f759bbef1efe2e8498ffff84e76d436/configs/projects/pxp-agent.rb#L27

Reply:
Going to work on debugging this today with @jelinwils

Reply:
The `vanagon dependencies pxp-agent <platform>` command needs to retrieve settings from the already built and promoted `puppet-runtime` as specified in:

```json
{"location":"http://builds.delivery.puppetlabs.net/puppet-runtime/202202020/artifacts/","version":"202202020"}
```

Presumably we don't have access to this in the context of a github action?

Reply:
you do, that's what the reverse proxy is there for

Reply:
@LivingInSyn there's a reverse proxy for `artifactory`, but not `builds`, right?

Reply:
The reverse proxy can access both. I'll file a bug for that failure condition that reports as a success and we'll get it fixed along with the `.snyk` support

Reply:
The reverse proxy can't resolve DNS for `builds`: https://github.com/puppetlabs/pxp-agent-vanagon/runs/5128148360?check_suite_focus=true#step:4:41

Reply:
you're missing this from your action:

https://github.com/puppetlabs/security-snyk-vanagon-action/blob/main/sample_workflow.yaml#L25

Without `urlsToReplace` the builds url is never replaced with the reverse proxy host

Reply:
This branch has the `urlsToReplace` and `newHost` keys, but running the action still fails: https://github.com/puppetlabs/pxp-agent-vanagon/runs/5101556125?check_suite_focus=true#step:4:41

Those two values are the same as the sample workflow in the action repo. Should those be different? I'm guessing that I need the reverse proxy URL in the `newHost` key (which is currently set to `localhost`)?

Reply:
@mhashizume the error you just referenced is a 401 coming from the reverse proxy:

https://github.com/puppetlabs/pxp-agent-vanagon/runs/5101556125?check_suite_focus=true#step:4:41

you are referencing the wrong secrets here:

https://github.com/puppetlabs/pxp-agent-vanagon/actions/runs/1797792732/workflow#L23-L24

The ones added to this repo are:

`SEC_PUBLIC_RPROXY_KEY` and `SEC_PUBLIC_RPROXY_USER`

Reply:
Thanks for catching that. I updated those variables in the workflow a little earlier today and seem to still be encountering the same issue from the reverse proxy.

You can see the correct variable names in this workflow: https://github.com/puppetlabs/pxp-agent-vanagon/actions/runs/1820101684/workflow

Still result in a 401 error: https://github.com/puppetlabs/pxp-agent-vanagon/runs/5130999380?check_suite_focus=true#step:4:42

Reply:
That's actually a `Error: 404` now, which is good, it's a _different_ error 😂  Lets setup some time tomorrow to just jam this out and get y'all moving

Reply:
actually, it successfully completed the following project+platforms
```
2022/02/09 19:48:01 running snyk on pxp-agent ubuntu-20.04-amd64
2022/02/09 19:48:01 testing  pxp-agent ubuntu-20.04-amd64
2022/02/09 19:48:01 running snyk on pxp-agent ubuntu-20.04-aarch64
2022/02/09 19:48:01 testing  pxp-agent ubuntu-20.04-aarch64
2022/02/09 19:48:01 running snyk on pxp-agent sles-15-x86_64
2022/02/09 19:48:01 testing  pxp-agent sles-15-x86_64
2022/02/09 19:48:01 running snyk on pxp-agent ubuntu-16.04-i386
2022/02/09 19:48:01 running snyk on pxp-agent solaris-11-sparc
2022/02/09 19:48:01 running snyk on pxp-agent solaris-11-i386
2022/02/09 19:48:01 testing  pxp-agent solaris-11-i386
2022/02/09 19:48:01 testing  pxp-agent ubuntu-16.04-i386
2022/02/09 19:48:01 testing  pxp-agent solaris-11-sparc
2022/02/09 19:48:06 finished snyk testing:  pxp-agent sles-15-x86_64
2022/02/09 19:48:06 Finished running snyk on pxp-agent sles-15-x86_64
2022/02/09 19:48:07 finished snyk testing:  pxp-agent ubuntu-20.04-aarch64
2022/02/09 19:48:07 Finished running snyk on pxp-agent ubuntu-20.04-aarch64
2022/02/09 19:48:08 finished snyk testing:  pxp-agent ubuntu-16.04-i386
2022/02/09 19:48:08 Finished running snyk on pxp-agent ubuntu-16.04-i386
2022/02/09 19:48:08 finished snyk testing:  pxp-agent ubuntu-20.04-amd64
2022/02/09 19:48:08 Finished running snyk on pxp-agent ubuntu-20.04-amd64
2022/02/09 19:48:08 finished snyk testing:  pxp-agent solaris-11-i386
2022/02/09 19:48:08 Finished running snyk on pxp-agent solaris-11-i386
2022/02/09 19:48:08 finished snyk testing:  pxp-agent solaris-11-sparc
2022/02/09 19:48:08 Finished running snyk on pxp-agent solaris-11-sparc
```
The others are 404's from artifactory or builds